tsconfig.json

```diff
@@ -0,0 +1,14 @@
+{
+  "include": ["frontend/**/*"],
+  "exclude": [],
+  "compilerOptions": {
+    "target": "esnext",
+    "module": "commonjs",
+    "declaration": true,
+    "outDir": "lib",
```

Comment:
Is this where we want the transpiled output to go?

Reply:
With this setup the compiled code is never emitted and this setting is ignored unless you were manually to run tsc. Since we’re still using lib for a few things, maybe we should change this to “null” or “trash” or “tsc_output.”

Reply:
I updated this to `unused_ts_output` which will hopefully convey the fact that it can be safely deleted.

Reply:
I'm a little confused then, probably because of lack of familiarity with this portion of the Shields codebase.

I assume we must have _something_ that's transpiling our frontend source (which will be TS after this is merged) into some browser-compatible JS. If we're not using the TS compiler to do that, is that being done via something else (babel perhaps?)

Reply:
Yes, you're right. This happens in three places:

- `gatsby dev` (handled by `gatsby-plugin-typescript` which is configured here: https://github.com/badges/shields/pull/3722/files#diff-b9e136416b90437fa1dac910280b45fc)
- `gatsby build` (same)
- Running tests, where it's handled by `ts-mocha`. I'm not sure which tools it might be using under the hood.

The dev output gets cached by Gatsby (I think in `.cache`) and the generated build goes into `public/` and generates [a tree that looks like this](https://github.com/badges/shields/tree/gh-pages). We definitely don't lint or run coverage on either of those. And since we never invoke `tsc`, the transpiled output is never written to the configured TypeScript `outDir`.

Does that make sense?

Reply:
> Does that make sense?

💯 indeed, thanks so much!